torch/testing/_internal/common_nn.py

```diff
@@ -5120,15 +5121,31 @@ def assert_module_parameters_are(tensor_type, device_id=None):
                 assert_module_parameters_are(torch.cuda.FloatTensor, 1)  # type: ignore[attr-defined]
         else:
             # check that float()/double() casters work correctly
+            def to_type(tensor, real, complex):
+                if tensor.is_complex():
+                    return tensor.to(complex)
+                elif tensor.is_floating_point():
+                    return tensor.to(real)
+                else:
+                    return tensor
+
+            def to_half(x):
```

Comment:
I don't think we should add `to_half` for complex tensors, since the support for `torch.complex32` is bare minimum right now. In fact, @mruberry and I both agreed we should probably disable `torch.complex32` for 1.8 release.